src/mgmt/tests/mocks.c

```diff
@@ -275,9 +275,32 @@ ds_mgmt_free_pool_list(struct mgmt_list_pools_one **poolsp, uint64_t len)
 int				ds_mgmt_set_prop_return;
 int
 ds_mgmt_pool_set_prop(uuid_t pool_uuid, daos_prop_t *prop,
-					  daos_prop_t **result)
+		      daos_prop_t **result)
 {
-	return ds_mgmt_set_prop_return;
+	daos_prop_t *res_prop;
+	size_t i;
+
+	if (prop == NULL)
```

Comment:
I think it's totally okay to let the unit tests crash and burn in this case. Production code won't touch this, and when I get a segfault in a unit test, the first thing I do is run gdb and inspect the stack trace. Discovery of coding errors is much quicker that way than trying to figure out the source of a surprising return code.

Reply:
Done.